zerver/lib/webhooks/common.py

```diff
@@ -114,3 +130,104 @@ def validate_extract_webhook_http_header(request: HttpRequest, header: str,
         raise MissingHTTPEventHeader(header)
 
     return extracted_header
+
+class ThirdPartyAPIAmbassador:
+    """ Using an instance of this class, integrations/webhook view code can systematically
+    authenticate with a third party API and then send multiple requests to it, store the results,
+    as well as set triggers to be run before and after the calls. The http_api_callback method of
+    this class is the method that integrations/webhook view code should use to actually call the
+    third party api endpoint. """
```

Comment:
I've pointed this out multiple times but please wrap your code at 80 characters everywhere.

Reply:
Got it :+1:, I've wrapped _**every**_ single line of code added in this PR at 80 chars, no excuses.